src/main/java/org/apache/commons/lang3/StringEscapeUtils.java

```diff
@@ -429,6 +429,65 @@ public StringEscapeUtils() {
       super();
     }
 
+    /**
+     * <p>Convenience wrapper for {@link java.lang.StringBuilder} providing escape methods.</p>
+     *
+     * <p>Example:</p>
+     * <pre>
+     * new Builder(ESCAPE_HTML4)
+     *      .append("&lt;p>")
+     *      .escape("This is paragraph 1 and special chars like &amp; get escaped.")
+     *      .append("&lt;/p>&lt;p>")
+     *      .escape("This is paragraph 2 &amp; more...")
+     *      .append("&lt;/p>")
+     *      .toString()
+     * </pre>
+     * 
+     * @since 3.6
+     */
+    public static class Builder {
+
+        private final StringBuilder sb;
+        private final CharSequenceTranslator translator;
+
+        public Builder(final CharSequenceTranslator translator) {
```

Comment:
I'd rather have a private constructor and then put a static factory method into `StringEscapeUtils` instead of having the inner class that one needs to construct like `new StringEscapeUtils.Builder(ESCAPE)`